scripts/setup_nlp.sh

```diff
@@ -0,0 +1,83 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+ENV=$1
+REGION=$2
+
+if [[ $ENV != "staging" && $ENV != "prod" && $ENV != "autopush" && $ENV != "dev" && $ENV != "feeding-america" && $ENV != "tidal" && $ENV != "karnataka" ]]; then
+  echo "First argument should be 'staging' or 'prod' or 'autopush' or 'dev' or 'feeding-america' or 'tidal' or 'karnataka'"
+  exit
+fi
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT="$(dirname "$DIR")"
+
+if [[ $REGION == "" ]]; then
+  REGION=$(yq eval '.region.primary' $ROOT/deploy/gke/$ENV.yaml)
+fi
+
+PROJECT_ID=$(yq eval '.project' $ROOT/deploy/gke/$ENV.yaml)
+PROJECT_NUMBER=$(gcloud projects list --filter="${PROJECT_ID}" --format="value(PROJECT_NUMBER)")
+
+MODEL_NAME=$(yq eval '.model_name' $ROOT/deploy/gke/$ENV.yaml)
+NETWORK_NAME=$(yq eval '.network_name' $ROOT/deploy/gke/$ENV.yaml)
+
+MODEL_BUCKET="gs://${PROJECT_ID}-modelstore"
+MODEL_PATH="${MODEL_BUCKET}/${MODEL_NAME}"
+
+if ! gsutil -q stat "${MODEL_PATH}/*" ; then
+  echo "Model path does not exists ${MODEL_PATH}"
+  
+  # This might fail, meaning that the bucket already exists.
+  gsutil mb -p "${PROJECT_ID}" -c STANDARD -l US -b on "${MODEL_BUCKET}"
+  
+  exit
+fi
+
+echo "Setting up Vertex AI model for ${MODEL_PATH}"
+
+MACHINE_TYPE="n1-highmem-2"
+VERTEX_MODEL_NAME="vxm-${MODEL_NAME}"
+VERTEX_ENDPOINT_NAME="vxe-${MODEL_NAME}"
+
+gcloud ai models upload \
+    --project="${PROJECT_ID}" \
+    --region="${REGION}" \
+    --display-name="${VERTEX_MODEL_NAME}" \
+    --artifact-uri="${MODEL_PATH}" \
+    --container-image-uri="us-docker.pkg.dev/vertex-ai-restricted/prediction/tensorflow-enterprise-inference-engine-alpha:latest" \
+    --container-args='--port=8500,--rest_api_port=8080,--model_name=default,--model_base_path=$(AIP_STORAGE_URI)' \
+    --container-ports=8080 \
+    --container-predict-route=/v1/models/default:predict \
+    --container-health-route=/v1/models/default
+
+gcloud beta ai endpoints create \
+    --project="${PROJECT_ID}" \
+    --region="${REGION}" \
+    --network="projects/${PROJECT_NUMBER}/global/networks/${NETWORK_NAME}" \
+    --display-name="${VERTEX_ENDPOINT_NAME}"
+
+ENDPOINT_ID=$(gcloud ai endpoints list --project="${PROJECT_ID}" --region="${REGION}" --filter "display_name=${VERTEX_ENDPOINT_NAME}" --format="value(ENDPOINT_ID.scope())")
+MODEL_ID=$(gcloud ai models list --project="${PROJECT_ID}" --region="${REGION}" --filter "display_name=${VERTEX_MODEL_NAME}" --format="value(MODEL_ID.scope())")
+
+gcloud beta ai endpoints deploy-model "${ENDPOINT_ID}" \
```

Comment:
Does this script contains one time setup or it is used for each new deployment. If it has one time setup, should move it to $ROOT/gke as "setup_nlp.sh". Then the remaining part can be renamed as "deploy_nlp.sh" since it's not really a setup?

Reply:
Moved it to `deploy_ai.sh` which is a more generic name.